x-pack/plugins/cloud_security_posture/server/plugin.ts

```diff
@@ -70,25 +61,17 @@ export class CspPlugin
     this.logger = initializerContext.logger.get();
   }
 
-  private readonly CspAppService = new CspAppService();
-
   public setup(
     core: CoreSetup<CspServerPluginStartDeps, CspServerPluginStart>,
     plugins: CspServerPluginSetupDeps
   ): CspServerPluginSetup {
-    const cspAppContext: CspAppContext = {
-      logger: this.logger,
-      service: this.CspAppService,
-      security: plugins.security,
-    };
-
     core.savedObjects.registerType(cspRuleAssetType);
     core.savedObjects.registerType(cspRuleTemplateAssetType);
 
-    const router = core.http.createRouter<CspRequestHandlerContext>();
-
-    // Register server side APIs
-    defineRoutes(router, cspAppContext);
+    setupRoutes({
+      core,
+      logger: this.logger,
+    });
```

Comment:
> **Note**
> SUBJECTIVE 😄 

calling `setupFoo` in our plugin's `setup` method is preferable to inlining everything there, as it's easier to reason about.